language-plutus-core/src/Language/PlutusCore/Pretty/Readable.hs

```diff
@@ -226,8 +227,9 @@ instance PrettyReadableBy configName (tyname a) =>
         TyApp _ fun arg         -> applicationDoc config fun arg
         TyVar _ name            -> unit $ prettyName name
         TyFun _ tyIn tyOut      -> arrowDoc config tyIn tyOut
-        TyFix _ name body       -> bind $ \bindBody ->
-            "fix" <+> prettyName name <> "." <+> bindBody body
+        -- TODO: add another combinator for doing this. Or fix the existing one somehow.
+        TyIFix _ pat arg        -> compoundDoc @_ @(Type tyname a) config juxtApp $ \_ juxtRight ->
```

Comment:
What does this do?

Reply:
The `compoundDoc` combinator from the readable pretty-printing machinery that knows about precedence and associativity of operators is not as generic as it should be. I tried to quickly fix that, but failed. I'll look into it later as it's unrelated to this PR.